src/sql/workbench/browser/scriptingActions.ts

```diff
@@ -50,7 +50,7 @@ export class ScriptExecuteAction extends Action {
 		super(id, label);
 	}
 
-	public run(actionContext: BaseActionContext): Promise<boolean> {
+	public async run(actionContext: BaseActionContext): Promise<boolean> {
```

Comment:
Since `script` returns a `Promise<void>` and this function returns a `Promise<boolean>`, you should do something like 
```
await script(...
return true

Reply:
script now returns a boolean

Reply:
https://github.com/microsoft/azuredatastudio/blob/amir/promise_cleanup_2/src/sql/workbench/browser/scriptingUtils.ts#L116-L161